<?xml version="1.0"?>
<!-- Provided by sleutene for convenience -->
<launch>
  <!-- Launch simulator. -->
  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find sjtu_drone)/launch/simple.launch">
    <arg name="world_name" value="$(find sjtu_drone)/worlds/skokloster_castle.world" />
  </include>

  <!-- Emulate controller parameters-->
  <group ns="ardrone_driver">
    <param name="euler_angle_max" value="1.0" /> <!-- original value: 0.21 -->
    <param name="control_vz_max" value="700" />
    <param name="control_yaw" value="1.75" />
  </group>

  <!-- Launch visualisation. -->
  <node pkg="rviz" type="rviz" name="rviz"
    args="-d $(find
  ardrone_practicals)/config/rviz_vi_ekf.rviz" />

  <!-- Launch estimator / control arp_node. -->
  <node name="arp_node" pkg="ardrone_practicals" type="arp_node"
    output="screen">
    <remap from="ardrone/front/image_raw" to="drone/front_camera/image_raw" />
    <remap from="ardrone/imu" to="drone/imu" />
    <remap from="ardrone/takeoff" to="drone/takeoff" />
    <remap from="ardrone/reset" to="drone/reset" />
    <remap from="ardrone/land" to="drone/land" />
    <remap from="ardrone/navdata" to="drone/navdata" />

    <!-- Camera parameters - from previous simulations -->
    <param name="fu" value="185.6909" />
    <param name="fv" value="185.6909" />
    <param name="cu" value="319.5" />
    <param name="cv" value="179." />
    <param name="k1" value="0" />
    <param name="k2" value="0" />
    <param name="p1" value="0" />
    <param name="p2" value="0" />


    <!-- Camera parameters - from localization -->
    <!-- <param name="fu" value="390.598" />
    <param name="fv" value="390.598" />
    <param name="cu" value="319.5" />
    <param name="cv" value="179." />
    <param name="k1" value="0" />
    <param name="k2" value="0" />
    <param name="p1" value="0" />
    <param name="p2" value="0" /> -->

    <param name="mapFocalLength" value="185.6909" />

    <rosparam param="T_SC">[0.0, 0.0, 1.0, 0.2, -1.0, 0.0, 0.0, 0.0, 0.0, -1.0, 0.0, 0.0, 0.0, 0.0,
      0.0, 1.0]</rosparam>

    <!-- Map parameters -->
    <param name="map" value="okvis2-slam-final_map_skokloster.csv" />
    <param name="occupancymap" value="occupancy-map_skokloster.dat" />

    <!-- arp_node parameters -->
    <param name="THRESHOLD_MIDDLE_WAYPOINTS" value="0.6" />
    <param name="THRESHOLD_LANDING" value="0.3" /> <!-- have a smaller tolerance for the last point -->
    <param name="FLYING_HEIGHT" value="1.4" />
    <param name="LAND_HEIGHT" value="0.3" />

    <!-- Frontend parameters -->
    <param name="maxPixelDistance" value="5" /> <!-- had 30 -->
    <param name="numPosesToMatch" value="30" /> <!-- had 15 -->

    <!-- Autopilot parameters - PID X -->
    <param name="pid_X_k_p" value="1.0" />
    <param name="pid_X_k_i" value="0.05" />
    <param name="pid_X_k_d" value="0.2" />

    <!-- Autopilot parameters - PID Y -->
    <param name="pid_Y_k_p" value="1.0" />
    <param name="pid_Y_k_i" value="0.05" />
    <param name="pid_Y_k_d" value="0.15" />

    <!-- Autopilot parameters - PID Z -->
    <param name="pid_Z_k_p" value="1.0" />
    <param name="pid_Z_k_i" value="0.02" />
    <param name="pid_Z_k_d" value="0.1" />

    <!-- Autopilot parameters - PID Yaw -->
    <param name="pid_Yaw_k_p" value="1.0" />
    <param name="pid_Yaw_k_i" value="0.02" />
    <param name="pid_Yaw_k_d" value="0.1" />

    <!-- BRISK parameters -->
    <param name="uniformityRadius" value="40" /> <!-- 10 -->
    <param name="octaves" value="2" /> <!-- 0 -->
    <param name="absoluteThreshold" value="50" /> <!-- 100 -->
    <param name="maxNumKpt" value="1000" /> <!-- 2000 -->

    <!-- point B position -->
    <rosparam param="pointB">[5.0, 10.0, 1.0]</rosparam>
  </node>

</launch>